File: JavaCivitas/src/civitas/SorpresaPorJugador.java
package civitas;

import java.util.ArrayList;

public class SorpresaPorJugador extends Sorpresa{

    private int valor;

    SorpresaPorJugador(int valor_a, String texto){

        super(texto);
        valor = valor_a;

    }

    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){

        if (jugadorCorrecto(actual, todos) && todos.get(actual) != null){
            super.informe(actual, todos);
            Sorpresa tipo1=new SorpresaPagarCobrar(valor*-1,"Decrementa saldo");
            Sorpresa tipo2=new SorpresaPagarCobrar(valor*todos.size(),"Incrementa saldo");
            for(int i=0;i<todos.size();i++){
                if(actual!=i){
                    tipo1.aplicarAJugador(i, todos);
                } else {
                    tipo2.aplicarAJugador(i, todos);
                }
            }
        }

    }

}

File: RubyCivitas/lib/sorpresa_porjugador.rb
module Civitas
  class Sorpresa_porjugador < Sorpresa
    
    def initialize(valor, texto)
      
      @valor = valor
      super(texto)
      
    end
    
    def aplicar_a_jugador(actual, todos)

      if jugador_correcto(actual, todos)
        informe(actual, todos)
        sorpresa_c = Sorpresa_pagarcobrar.new(@valor*(-1),"Decrementa saldo")
        sorpresa_p = Sorpresa_pagarcobrar.new(@valor*todos.size,"Incrementa saldo")

        for i in 0...todos.size
          if actual!=i
            sorpresa_c.aplicar_a_jugador(i,todos)
          else
            sorpresa_p.aplicar_a_jugador(i,todos)
          end
        end
        
      end
    end
    
  end
end
